### PR TITLE
 Call click URL when clients follow a link from the sponsored homepage shelf

### LINKED ISSUE #9721: Call click URL when clients follow a link from the sponsored homepage shelf
When implementing #9719 to fetch the sponsored shelf through the specific API we're building in mozilla/addons-server#15617 we need to call a special click URL when the user follows a link from the shelf. See https://github.com/mozilla/addons-server/issues/15619 for the implementation of that endpoint.

We may want to track conversions (installs following a click) as well through Adzerk - that's up to product, we'd need to research what needs to be implemented for that to happen.

### PR DESCRIPTION
**Pull request by bobsilverberg**
~~Depends on #9760~~
Depends on https://github.com/mozilla/addons-server/pull/15768
Fixes #9721

### CODE REVIEW

**Comment by bobsilverberg** [comment]
Note that this includes commits from 2 other PRs on which this is built, so it will probably be simpler to review this after those other two PRs have been merged.

**Review by eviljeff — APPROVED** [review]
seems reasonable but I've only looked at the last commit and I expect you'll have to rebase once the other patches are merged

**Comment by bobsilverberg** [comment]
This has been rebased. @willdurand I think you said you wanted to take a look at it.

**Comment by bobsilverberg** [comment]
This is going to have to change as a result of https://github.com/mozilla/addons-server/pull/15768, so let's wait for that to land, and then I'll update this to use the new `event` endpoint.

**Comment by bobsilverberg** [comment]
I've updated this to address the changes in https://github.com/mozilla/addons-server/pull/15768, so you might want to take another quick look, @eviljeff.

@willdurand I think you said you wanted to look at this one too.

File: src/amo/components/SponsoredAddonsShelf/index.js
Comment (eviljeff) [inline src/amo/components/SponsoredAddonsShelf/index.js:63]: when will `type` be missing?  it's the key of the `event_data` dict/object where the data value came from.

Comment (bobsilverberg) [reply src/amo/components/SponsoredAddonsShelf/index.js:63]: When we call `formatDataForBeacon` in order to send a beacon for `impression_url` (i.e., to `/shelves/sponsored/impression/`), we don't need a `type`.

Comment (eviljeff) [reply src/amo/components/SponsoredAddonsShelf/index.js:63]: oh, it's used for impressions too.  I didn't catch that.

File: src/amo/components/SponsoredAddonsShelf/index.js
Comment (willdurand) [inline src/amo/components/SponsoredAddonsShelf/index.js:27]: Why is this needed?

**Review by willdurand — COMMENTED** [review]
It looks good to me (even if I'd like explanations about the absolute `EVENT_URL`, which is not something we usually do).

One thing, though: I am wondering if we shouldn't take the DNT setting into account for click beacons because that's tracking I believe.

**Comment by willdurand** [comment]
> One thing, though: I am wondering if we shouldn't take the DNT setting into account for click beacons because that's tracking I believe.

cc @diox

File: src/amo/components/SponsoredAddonsShelf/index.js
Comment (bobsilverberg) [reply src/amo/components/SponsoredAddonsShelf/index.js:27]: I guess I was just trying to reuse logic we already have for calculating an API URL. I can change it to 
```
`${config.get('apiPath')}${config.get('apiVersion')}/shelves/sponsored/event/`
```

to keep it relative.

**Comment by bobsilverberg** [comment]
> It looks good to me (even if I'd like explanations about the absolute `EVENT_URL`, which is not something we usually do).
> 
> One thing, though: I am wondering if we shouldn't take the DNT setting into account for click beacons because that's tracking I believe.

Yeah, that's a good point. We probably should do that the same we we do with our GA tracking code. It could potentially skew our numbers from reality, but I don't imagine the effect will be that large. I think this is something that @jvillalobos needs to chime in on as well.

**Comment by eviljeff** [comment]
if this shelf is still behind a feature flag I think it's fine to merge and sort out DNT afterwards in a follow-up issue.

**Comment by bobsilverberg** [comment]
> if this shelf is still behind a feature flag I think it's fine to merge and sort out DNT afterwards in a follow-up issue.

Good point. I can open a follow-up about that.

File: src/amo/components/SponsoredAddonsShelf/index.js
Comment (bobsilverberg) [reply src/amo/components/SponsoredAddonsShelf/index.js]: FTR, I have updated the code to use the above.

File: src/core/api/index.js
Comment (willdurand) [inline src/core/api/index.js:23]: Is this still needed?

**Review by willdurand — APPROVED** [review]
r+wc

**Comment by willdurand** [comment]
> > if this shelf is still behind a feature flag I think it's fine to merge and sort out DNT afterwards in a follow-up issue.
> 
> Good point. I can open a follow-up about that.

wfm

File: src/core/api/index.js
Comment (bobsilverberg) [reply src/core/api/index.js]: Nope! Fixed!

**Pull request merged**
